python/lsst/daf/persistence/butlerHelpers.py

```diff
@@ -0,0 +1,80 @@
+#
+# LSST Data Management System
+# Copyright 2008-2018 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+"""Common tasks and idioms performed with the Butler.
+"""
+
+__all__ = ["dataExists", "searchDataRefs"]
+
+
+def searchDataRefs(butler, datasetType, level="", dataId=None):
+    """Find all data references for a partial data ID.
+
+    Parameters
+    ----------
+    butler: `lsst.daf.persistence.Butler`
+        The repository to query for data.
+    datasetType : `str`
+        The type of data references to return.
+    level : `str`
+        The level of data ID at which to search. If the empty string, the
+        default level for ``datasetType`` shall be used.
```

Comment:
Please reference wherever it is in the butler docs that `level` is better defined. If anywhere.

Reply:
I'm not aware of any definition of `level` in the docs.

Reply:
I suggest we leave it alone. It was necessary in the butler when I wrote it, because of dataset types such as raw images, which were per amplifier. I strongly suspect it's still necessary. It will go away with gen3 butler, thank heavens. I hated the code when I wrote it and I still hate it.